### PR TITLE
CAP CI: implement pools

### PR DESCRIPTION
**Pull request by prabalsharma**
## Description

- Implement pools and remove ekcp
CI now uses concourse pool resource. Inorder to test you cluster you will have to place you kubeconfig in ${backend}-kube-hosts branch of git@github.com:SUSE/cf-ci-pools.git in unclaimed folder.

- Fix triggers
This is a bug fix. Post deploy, CI's consequent jobs  were getting triggered by kubecf git resource. Now all jobs gets triggered via kubeconfig / pool resource

- Remove kubecf resource
kubecf github repo is not required in this CI as a resource.

- remove kubecf CI to avoid any confusion

## Motivation and Context
Pool implementation  is required to manage locks on kubeconfigs while testing.

## How Has This Been Tested?
https://concourse.suse.dev/teams/main/pipelines/prabal-demo-release?group=eks

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code has security implications.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by f0rmiga** [comment]
@prabalsharma You need to sign the CLA.

File: .concourse/cap-release/pipeline.template [hunk 1]
```diff
@@ -38,12 +38,6 @@ resource_types:
     tag: release
 
 resources:
-- name: kubecf-master
```
Comment (viccuad) [inline .concourse/cap-release/pipeline.template]: Top! :100:

**Comment by viccuad** [comment]
Many thanks, wooho! :tada:

File: .concourse/cap-release/pipeline.template [hunk 2]
```diff
@@ -59,13 +53,25 @@ resources:
...
+{{end}}
+
 {{ $pipelineName := .Env.PIPELINE }}
 
-{{- $obtain_ekcp := `
-            # obtain kubeconfig from ekcp
+{{- $obtain_kubeconfig := `
+            # obtain kubeconfig from pool
+            pool_file=${pool_file:-pool.kube-hosts/metadata}
+            cp ${pool_file} catapult/kubeconfig_$CLUSTER_NAME
```
Comment (viccuad) [inline .concourse/cap-release/pipeline.template]: It is possible to also obtain the cluster name from the name of the pool file: `pool.kube-hosts/name`.
That way the following jobs can just depend on `input: pool.kube-hosts`.

On the kubecf pipeline:
https://github.com/cloudfoundry-incubator/kubecf/blob/2d6f7c19f4540cb7154bdebe17794e2232e30fab/.concourse/pipeline.yaml.gomplate#L122-L123

and then just keep doing what you are doing on this PR, use `input: pool.kube-hosts` and compute CLUTER_NAME and KUBECFG values.

That way we accept whatever name the cluster has on the pool, and we can remove all hardcoded CLUSTER_NAME: 
https://github.com/cloudfoundry-incubator/kubecf/blob/b3163dbee4b6117365d46420e8c79b1e0c2f763f/.concourse/cap-release/pipeline.template#L201

File: .concourse/cap-release/pipeline.template [hunk 3]
```diff
@@ -59,13 +59,25 @@ resources:
...
 
+
+# Pool resource with kube cluster information
+{{ range $_, $Backend := (ds "BACKEND").backend -}} # for all backends
+- name: {{$Backend}}-pool.kube-hosts
+  type: pool
+  source:
+    uri: git@github.com:SUSE/cf-ci-pools.git
+    private_key: ((github-private-key))
+    branch: {{$Backend}}-kube-hosts
```
[the remark this answers is not in the document]
Comment (prabalsharma) [reply .concourse/cap-release/pipeline.template]: the convention has been to use {ecp/gke/eks}-kube-hosts the reason being pools should not be dependent on which CI it is being used by. Same pools can be used by a local (running on anyones machine) CI or a provo CI or van CI.

**Comment by prabalsharma** [comment]
@viccuad once done with review, please ping Vlad so that he can merge this PR overriding CLA requirement. There is some issue with CLA whitelisting for me.

**Comment by viccuad** [comment]
I miss using concourse queue instead of the pool resource queue, as the concourse queue never times out. That way full pipeline runs can be queued, and once a run finishes, the next one starts (with possible different inputs, pins , etc). This makes it more automated, no need to trigger runs manually, you queue 5 full pipeline runs, and they will run overnight once there are available clusters. For that we need:

https://github.com/cloudfoundry-incubator/kubecf/blob/master/.concourse/pipeline.yaml.gomplate#L367 

And to extend the timeout of obtaining the first lock for the cluster to 4h, which is the full duration of the pipeline:

https://github.com/cloudfoundry-incubator/kubecf/blob/b6ea05538f58b0cefd5075379299336ffcf2d8fb/.concourse/pipeline.yaml.gomplate#L374

 
Nevertheless, it works, so it's not a merge blocker for me. We can improve it in following iterations.

**Comment by viccuad** [comment]
As requested, @viovanov can you merge this PR overriding CLA requirements? Prabal has problems with that.

**Pull request merged**